src/middleware/createApiMiddleware.js

```diff
@@ -54,13 +54,13 @@ function createMiddleware(host, defaultHeaders = getDefaultHeaders()) {
 
   const requestAction = async (method, { resource, params, headers } = {}) => {
     const url = getURL(resource, params);
+    const mergedHeaders = Object.assign({}, getDefaultHeaders(), defaultHeaders, headers);
```

Comment:
This can be done way easier inline around line 62 using the spread operator. In fact, I'm kinda face-palming myself for not doing this in the first place...

Give this a shot on line 62:

```js
headers: {
  ...getDefaultHeaders(),
  ...defaultHeaders,
  ...headers,
}
```